src/Parser/Printer/PrettyPrinterConfiguration.php

```diff
@@ -0,0 +1,102 @@
+<?php
+declare(strict_types=1);
+
+
+namespace Helmich\TypoScriptParser\Parser\Printer;
+
+use InvalidArgumentException;
+use LogicException;
+use Webmozart\Assert\Assert;
+
+/**
+ * PrinterConfiguration
+ *
+ * @package    Helmich\TypoScriptParser
+ * @subpackage Parser\PrettyPrinterConfiguration
+ */
+final class PrettyPrinterConfiguration
+{
+    /**
+     * @var string
+     */
+    public const INDENTATION_STYLE_SPACES = 'spaces';
+
+    /**
+     * @var string
+     */
+    public const INDENTATION_STYLE_TABS = 'tabs';
+
+    /**
+     * @var array
+     */
+    private const ALLOWED_INDENTATION_STYLES = [self::INDENTATION_STYLE_TABS, self::INDENTATION_STYLE_SPACES];
+
+    /**
+     * @var bool
+     */
+    private $addClosingGlobal;
+
+    /**
+     * @var bool
+     */
+    private $includeEmptyLineBreaks;
+
+    /**
+     * @var int
+     */
+    private $indentationSize;
+
+    /**
+     * @var string
+     */
+    private $indentationStyle;
+
+    public function __construct(bool $addClosingGlobal, bool $includeEmptyLineBreaks, int $indentationSize, string $indentationStyle)
+    {
+        if(!in_array($indentationStyle, self::ALLOWED_INDENTATION_STYLES)) {
+            throw new InvalidArgumentException(
+                sprintf('Indentation style must be one of %s but got %s',
+                    implode(',', self::ALLOWED_INDENTATION_STYLES),
+                    $indentationStyle
+                )
+            );
+        }
+
+        if($indentationSize > 0 && $indentationStyle === self::INDENTATION_STYLE_TABS) {
```

Comment:
I'm wondering if we can find a more intuitive interface for this -- for me, it's not really self-evident here that `$indentationSize` needs to be `>0` when using spaces, and `==0` when using tabs. Maybe we can add mutator functions that abstract these intricacies (and maybe even with a private constructor)? Something like...

```php
public function withSpaceIndentation(int $size): self {
    $c = clone $this;
    $c->indentationStyle = self::INDENTATION_STYLE_SPACES;
    $c->indentationSize = $size;
    return $c;
}

public function withTabIndentation(): self {
    $c = clone $this;
    $c->indentationStyle = self::INDENTATION_STYLE_TABS;
    $c->indentationSize = 1;
    return $c;
}
```

To be used (for example) as follows:

```php
$printer = new PrettyPrinter(
    PrettyPrinterConfiguration::getDefault()
        ->withSpaceIndentation(2)
);
```

Reply:
Sounds reasonable.